src/charm.py

```diff
@@ -94,19 +119,29 @@ def _on_install(self, event):
         else:
             self.unit.status = BlockedStatus("Missing token or org/repo path config")
 
-    def _on_upgrade_charm(self, event):
+    def _on_upgrade_charm(self, event: UpgradeCharmEvent):
+        """Handle the update of charm.
+
+        Args:
+            event (UpgradeCharmEvent): Event of charm upgrade.
+        """
         RunnerManager.install_deps()
 
-    def _on_config_changed(self, event):
-        self._ensure_event_timer("update-runner-bin", self.config["update-interval"])
-        self._ensure_event_timer("reconcile-runners", self.config["reconcile-interval"])
+    def _on_config_changed(self, event: ConfigChangedEvent):
+        """Handle the configuration change.
+
+        Args:
+            event (ConfigChangedEvent): Event of configuration change.
+        """
+        self._event_timer.ensure_event_timer("update-runner-bin", self.config["update-interval"])
+        self._event_timer.ensure_event_timer(
+            "reconcile-runners", self.config["reconcile-interval"]
+        )
 
         if self.config["path"] != self._stored.path:
-            prev_runner_manager = self._get_runner_manager(path=self._stored.path)
+            prev_runner_manager = self._get_runner_manager(path=str(self._stored.path))
```

Comment:
Why do you need this cast?

Reply:
For mypy type checking.
The `_get_runner_manager` method takes a `Optional[str]` but the `self._stored.path is `ObjectEvents` type.
Due to `self._stored` is `StoredState` object.

This mypy error shows up without casting to str:
```
static run-test: commands[0] | mypy /home/user/work/charm/github-runner-operator/src/ /home/user/work/charm/github-runner-operator/tests/
src/charm.py:144: error: Argument "path" to "_get_runner_manager" of "GithubRunnerOperator" has incompatible type overloaded function; expected "Optional[str]"  [arg-type]
```

Reply:
Thx for the explanation